wasm/src/encoding.c

```diff
@@ -316,3 +316,23 @@ opa_value *opa_json_marshal(opa_value *a)
 
     return opa_string_allocated(v, strlen(v));
 }
+
+OPA_BUILTIN
+opa_value *opa_json_is_valid(opa_value *a)
+{
+    if (opa_value_type(a) != OPA_STRING)
+    {
+        return NULL;
+    }
+
+
+    opa_string_t *s = opa_cast_string(a);
+    opa_value *r = opa_json_parse(s->v, s->len);
+
+	if (r == NULL)
+	{
+		return opa_boolean(false);
+	}
+
+	return opa_boolean(true);
```

Comment:
[nit] I'd probably go with
```suggestion
	return opa_boolean(r != NULL);
```

but it really doesn't matter. I guess it's all the same to the compiler.

Reply:
Thinking about this more, maybe a free wouldn't hurt?
```suggestion
	if (r == NULL)
	{
		return opa_boolean(false);
	}
    opa_free(r);
	return opa_boolean(true);
```

WDYT? It's been a while since I've played with those parts of the code base 🤔

Reply:
I like both suggestions. The latter probably don't hurt, as it did not break the tests at least, so I went with that. After all, we do not need the content of r, so we can free it.

I am not too familiar with the GC of WASM/C though.